src/cis/fam_cis_server.cpp

```diff
@@ -457,6 +457,47 @@ ::grpc::Status Fam_CIS_Server::copy(::grpc::ServerContext *context,
     return ::grpc::Status::OK;
 }
 
+::grpc::Status
+Fam_CIS_Server::backup(::grpc::ServerContext *context,
+                       const ::Fam_Backup_Restore_Request *request,
+                       ::Fam_Backup_Restore_Response *response) {
+    CIS_SERVER_PROFILE_START_OPS()
+    try {
+
+        famCIS->backup(request->regionid(), request->offset(),
+                       request->memserver_id(), request->filename(),
+                       request->uid(), request->gid(), request->size());
+    }
+
+    catch (Fam_Exception &e) {
+        response->set_errorcode(e.fam_error());
+        response->set_errormsg(e.fam_error_msg());
+        return ::grpc::Status::OK;
```

Comment:
CIS_SERVER_PROFILE_END_OPS in case of an error, or is it intentional ? Do look at other places if we do it in case of error.